gitlabform/gitlabform/processors/project/environments_processor.py

```diff
@@ -0,0 +1,64 @@
+import logging
+import json
+import operator
+
+from gitlabform.gitlab import GitLab
+from gitlabform.gitlabform.processors.abstract_processor import AbstractProcessor
+
+
+class EnvironmentsProcessor(AbstractProcessor):
+    def __init__(self, gitlab: GitLab):
+        super().__init__("environments")
+        self.gitlab = gitlab
+
+    def _process_configuration(self, project_and_group: str, configuration: dict):
+        logging.debug(
+            "Environments BEFORE: %s",
+            self.gitlab.get_all_environments(project_and_group),
+        )
+        e = self.gitlab.get_all_environments(project_and_group)
```

Comment:
Nit: single-letter variables are not great except maybe iterators in loops as they are very uninformative and can be misleading. When I was reading the code below guessed that `e` is a single environment and I was wrong...

Reply:
Also when debug output will be enabled then this line will do 2nd call to get all the environments to GitLab in a row. Check out https://github.com/egnyte/gitlabform/blob/v2/gitlabform/gitlabform/processors/project/project_push_rules_processor.py#L17-L19 where we do it only once.